students/rgpag/lesson03/slicingLab.py

```diff
@@ -0,0 +1,52 @@
+def exchange_first_last(seq):
+	""" return seq with first and last elements exchanged"""
+	L=len(seq)
+	return seq[-1]+seq[1:L-1]+seq[0]
+
+	a_string = "this is a string"
+	a_tuple = (1,2,3,4,5,6)
+
+	assert exchange_first_last(a_string) == "ghis is a strint"
+	assert exchange_first_last(a_tuple) == (6,2,3,4,5,1)
+
+def rem_odd(seq):
+	"""return seq dropping odd elements"""
+	return seq[::2]
+
+	a_string = "this is a string"
+	a_tuple = (1,2,3,4,5,6)
+
+	assert rem_odd(a_string) == 'ti sasrn'
+	assert rem_odd(a_tuple) == (1,3,5)
+
+def rem_ends_odds(seq):
+	"""return seq with first and last 4 elements dropped, and dropping odd elements of middle"""
+	temp_seq=seq[4:-4]
```

Comment:
This could be done as a one liner without the need for the other function `seq[4:-4:2]`